### PR TITLE
Update material of technical writing lecture 

### PR DESCRIPTION
**Pull request by uekerman**
## Description

- Tweaked minor things in the lecture, tried to shorten where meaningful
- Added an example based on the 2D diffusion code. My plan is to reuse this example next week in the tools section.

@IshaanDesai Please already check for consistency. I might tweak a few minor things still. 

## Checklist

- [x] I made sure that the markdown files are formatted properly.
- [x] I made sure that all hyperlinks are working.

- [x] I added the new content to the `README.md` inside the topics subdirectory, e.g., `01_version_control/README.md`.
- [x] I added the new content to the `timetable.md` in the root of this repository.

### CODE REVIEW

File: 04_documentation/technical_writing_slides.md [hunk 1]
```diff
@@ -97,9 +106,9 @@ Much content of this lecture is taken from [Write the Docs](https://www.writethe
 ## Technical Writing in Research Software
 
 - Like in a startup, **no dedicated technical writer in team** (even for large projects). Everybody does everything.
-    - -> You need writing skills.
-- In research, we anyway write a lot (papers, proposals, lecture material, ...).
-    - -> Writing/communication skills are very important anyway.
+    - --> You need writing skills.
+- In research, we write a lot (papers, proposals, lecture material, ...).
```
Comment (IshaanDesai) [inline 04_documentation/technical_writing_slides.md:110]:
```suggestion
- In research, one writes a lot (papers, proposals, lecture material, ...).
```
I think we can avoid *we* where is may sound like *we the researchers, and you the students* as a differentiation.

**Review by IshaanDesai — APPROVED** [review]
Looks good :+1: I was just wondering why we need to copy the diffusion code over here as an example when the original packaging exercise repo still exists: https://github.com/Simulation-Software-Engineering/diffusion2d ? You could create a branch in that repo titled `documentation` and add documentation there.

**Comment by uekerman** [comment]
> I was just wondering why we need to copy the diffusion code over here as an example when the original packaging exercise repo still exists: https://github.com/Simulation-Software-Engineering/diffusion2d ? You could create a branch in that repo titled documentation and add documentation there.

That's a good point I also struggled a bit with. I think the current situation is not optimal. We sometimes keep examples close to the lecture material and sometimes in separate repositories. IMO, the latter is nicer if it were one repo per code, which we could use for all demos (packaging, Docker, docs, testing). Currently, however, we use a different repo for each example.
I am not yet sure which code to use for the docs tools demo. Likely, I need a different version than what I used here. That's the reason why I added the code here.
Long story short: I think we should carefully refactor (directly) after the semester.
